Formula/redex.rb

```diff
@@ -0,0 +1,30 @@
+class Redex < Formula
+  desc "Bytecode optimizer for Android apps"
+  homepage "http://fbredex.com"
+  url "https://github.com/facebook/redex/archive/v1.0.0.tar.gz"
+  sha256 "58a41e7fb224c74438431d579f3ffa0ea7758d07730361d8a06d2c9e75d803a2"
+
+  head "https://github.com/facebook/redex.git"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "python3" => :build
+  depends_on "boost" => :build
+  depends_on "double-conversion" => :build
+  depends_on "gflags" => :build
+  depends_on "glog" => :build
+  depends_on "libevent" => :build
+  depends_on "jsoncpp" => :build
+
+  def install
+    system "autoreconf", "-ivf"
+    system "./configure", "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/redex", "-h"
```

Comment:
Given that the program's function is to take a binary as input, I'm not sure a more substantial test can be done without downloading a binary that isn't included with the application package. This could be possibly improved by adding a match to ensure the program runs and outputs it's help successfully after building.

The current output of the help is: 
```
usage: redex.py [-h] [-o [OUT]] [-j JARPATHS] [--redex-binary [REDEX_BINARY]]
                [-c CONFIG] [--sign] [-s [KEYSTORE]] [-a [KEYALIAS]]
                [-p [KEYPASS]] [-u] [-w [WARN]] [-d] [-m [PROGUARD_MAP]]
                [-q [PRINTSEEDS]] [-P PROGUARD_CONFIGS] [-k [KEEP]]
                [-S PASSTHRU] [-J PASSTHRU_JSON]
                input_apk

Given an APK, produce a better APK!

positional arguments:
  input_apk             Input APK file

optional arguments:
  -h, --help            show this help message and exit
  -o [OUT], --out [OUT]
                        Output APK file name (defaults to redex-out.apk)
  -j JARPATHS, --jarpath JARPATHS
                        Path to dependent library jar file
  --redex-binary [REDEX_BINARY]
                        Path to redex binary
  -c CONFIG, --config CONFIG
                        Configuration file
  --sign                Sign the apk after optimizing it
  -s [KEYSTORE], --keystore [KEYSTORE]
  -a [KEYALIAS], --keyalias [KEYALIAS]
  -p [KEYPASS], --keypass [KEYPASS]
  -u, --unpack-only     Unpack the apk and print the unpacked directories,
                        don't run any redex passes or repack the apk
  -w [WARN], --warn [WARN]
                        Control verbosity of warnings
  -d, --debug           Unpack the apk and print the redex command line to run
  -m [PROGUARD_MAP], --proguard-map [PROGUARD_MAP]
                        Path to proguard mapping.txt for deobfuscating names
  -q [PRINTSEEDS], --printseeds [PRINTSEEDS]
                        File to print seeds to
  -P PROGUARD_CONFIGS, --proguard-config PROGUARD_CONFIGS
                        Path to proguard config
  -k [KEEP], --keep [KEEP]
                        Path to file containing classes to keep
  -S PASSTHRU           Arguments passed through to redex
  -J PASSTHRU_JSON      JSON-formatted arguments passed through to redex
```